### PR TITLE
feat: federated /workflow_runs/{args.workflowRunId}/zip_link.json

### PR DESCRIPTION
**Pull request by saq7**
# Pull Request

## JIRA Ticket

[CZID-8507](https://czi-tech.atlassian.net/browse/CZID-8507)

## Description

This is a request, federates the `/zip_link.json` endpoint.

It converts the header based location response to a regular body response.

*Please tag a teammate as primary reviewer.*

## Notes

*Optional observations, comments or explanations.*
*If implementing a new feature, note any relevant analytics added.*

## Tests

* *Server-side code should have automated tests*
* *Client-side code and scripts should have at least a manual test plan*
Testing for this and point can be conducted using the graphiQL interface.

[CZID-8507]: https://czi-tech.atlassian.net/browse/CZID-8507?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: resolvers.ts [hunk 1]
```diff
@@ -263,6 +264,19 @@ export const resolvers: Resolvers = {
...
     },
+    ZipLink: async (root, args, context, info) => {
+      const res = await getFullResponse(`/workflow_runs/${args.workflowRunId}/zip_link.json`, args, context)
+      if (res.status !== 302) {
+        return {
+          url: null,
+          error: res.statusText,
+        };
+      }
+      const url = res.headers.get("location");
```
Comment (rainandbare) [inline resolvers.ts:275]: Why is this not `res.url`?

File: resolvers.ts [hunk 2]
```diff
@@ -263,6 +264,19 @@ export const resolvers: Resolvers = {
...
       return return_obj;
     },
+    ZipLink: async (root, args, context, info) => {
+      const res = await getFullResponse(`/workflow_runs/${args.workflowRunId}/zip_link.json`, args, context)
+      if (res.status !== 302) {
+        return {
+          url: null,
+          error: res.statusText,
+        };
+      }
```
Comment (rainandbare) [inline resolvers.ts:274]: So even if the response is 200 you want to deliver `null` as the url field?

**Review by rainandbare — COMMENTED** [review]
I tried the query on my GraphiQL interface and I wasn't able to make it work as is. I might be missing some context though.

File: resolvers.ts (hunk 2 above)
Comment (saq7) [reply resolvers.ts:274]: Yes, in this case. A 200 will actually represent an error given how the rails endpoint is set up. It returns a 302 with the url in the header.

File: resolvers.ts (hunk 1 above)
Comment (saq7) [reply resolvers.ts:275]: We don't need the url of the request in this case. We need the presigned s3 url that was returned in the header.

Comment (rainandbare) [reply resolvers.ts:275]: Ah Interesting. As it stands I am getting this response from a valid workflowRunId: 
`{
  "data": {
    "ZipLink": {
      "url": null,
      "error": "OK"
    }
  }
}`

I see in the `res` object from `getFullResponse` the appropriate s3 url and can access it by console logging or outputting `res.url`. 

What would be the best way to adjust the code to make a successful call?

**Comment by jfoo1984** [comment]
Could we fill out the PR description to provide context on what this PR is for and how we will test it?

**Review by rainandbare — APPROVED** [review]

**Pull request merged**